src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -497,8 +497,14 @@ private static bool RemoveEmptyDirectory(string fullPath, bool topLevel = false,
                     case Interop.Error.EACCES:
                     case Interop.Error.EPERM:
                     case Interop.Error.EROFS:
+                        // Prefer throwing DirectoryNotFoundException over UnauthorizedAccess IOException.
```

Comment:
Does this really should apply to all three cases: EACCES, EPERM, and EROFS?

Reply:
Before https://github.com/dotnet/runtime/pull/59520 it actually applied to any error because we checked if the directory exists before removing it.